sklearn/preprocessing/tests/test_polynomial.py

```diff
@@ -449,37 +449,145 @@ def test_spline_transformer_n_features_out(n_knots, include_bias, degree):
 
-        P_test = PolynomialFeatures(deg, include_bias=False).fit_transform(X)
-        assert_array_almost_equal(P_test, P[:, 1:])
 
-    interact = PolynomialFeatures(2, interaction_only=True, include_bias=True)
-    X_poly = interact.fit_transform(X)
-    assert_array_almost_equal(X_poly, P2[:, [0, 1, 2, 4]])
+@pytest.mark.parametrize(
+    "degree, include_bias, interaction_only, indices",
+    [
+        (3, True, False, slice(None, None)),
+        (3, False, False, slice(1, None)),
+        (3, True, True, [0, 1]),
+        (3, False, True, [1]),
+        ((2, 3), True, False, [0, 2, 3]),
+        ((2, 3), False, False, [2, 3]),
+        ((2, 3), True, True, [0]),
+        ((2, 3), False, True, []),
+    ]
+)
+@pytest.mark.parametrize(
+    "sparse_X",
+    [False, sparse.csr_matrix, sparse.csc_matrix],
+)
+def test_polynomial_features_one_feature(
+    single_feature_degree2,
+    degree,
+    include_bias,
+    interaction_only,
+    indices,
+    sparse_X,
+):
+    """Test PolynomialFeatures on single feature up to degree 3."""
+    X, P = single_feature_degree2
+    if sparse_X:
+        X = sparse_X(X)
+    tf = PolynomialFeatures(
+        degree=degree,
+        include_bias=include_bias,
+        interaction_only=interaction_only
+    ).fit(X)
+    out = tf.transform(X)
+    if sparse_X:
+        out = out.toarray()
+    assert_allclose(out, P[:, indices])
+    if tf.n_output_features_ > 0:
+        assert tf.powers_.shape == (tf.n_output_features_, tf.n_features_in_)
+
+
+@pytest.fixture()
+def two_features_degree3():
+    X = np.arange(6).reshape((3, 2))
+    x1 = X[:, :1]
+    x2 = X[:, 1:]
+    P = np.hstack(
+        [
+            x1 ** 0 * x2 ** 0,   # 0
+            x1 ** 1 * x2 ** 0,   # 1
+            x1 ** 0 * x2 ** 1,   # 2
+            x1 ** 2 * x2 ** 0,   # 3
+            x1 ** 1 * x2 ** 1,   # 4
+            x1 ** 0 * x2 ** 2,   # 5
+            x1 ** 3 * x2 ** 0,   # 6
+            x1 ** 2 * x2 ** 1,   # 7
+            x1 ** 1 * x2 ** 2,   # 8
+            x1 ** 0 * x2 ** 3,   # 9
+        ]
+    )
+    return X, P
+
 
-    assert interact.powers_.shape == (interact.n_output_features_,
-                                      interact.n_features_in_)
+@pytest.mark.parametrize(
+    "degree, include_bias, interaction_only, indices",
+    [
+        (2, True, False, slice(0, 6)),
+        (2, False, False, slice(1, 6)),
+        (2, True, True, [0, 1, 2, 4]),
+        (2, False, True, [1, 2, 4]),
+        ((2, 2), True, False, [0, 3, 4, 5]),
+        ((2, 2), False, False, [3, 4, 5]),
+        ((2, 2), True, True, [0, 4]),
+        ((2, 2), False, True, [4]),
+        (3, True, False, slice(None, None)),
+        (3, False, False, slice(1, None)),
+        (3, True, True, [0, 1, 2, 4]),
+        (3, False, True, [1, 2, 4]),
+        ((2, 3), True, False, [0, 3, 4, 5, 6, 7, 8, 9]),
+        ((2, 3), False, False, slice(3, None)),
+        ((2, 3), True, True, [0, 4]),
+        ((2, 3), False, True, [4]),
```

Comment:
For the sake of completeness and if I am not mistaken:

```suggestion
        ((2, 3), False, True, [4]),
        ((3, 3), True, False, [0, 6, 7, 8, 9]),
        ((3, 3), False, False, [6, 7, 8, 9]),
        ((3, 3), True, True, [0]),
        ((3, 3), False, True, []),  # would need 3 input features
```

I am not sure if the last case would not raise an error. 

Note that the case `degree=(3, 3), interaction_only=True` with 3 input features is also indirectly tested in `test_polynomial_feature_names`.